src/lib.rs

```diff
@@ -212,6 +214,49 @@ impl FromStr for Message {
     }
 }
 
+#[cfg(feature = "serde")]
+impl Serialize for Message {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        serializer.serialize_str(self.to_string().as_str())
+    }
+}
+
+#[cfg(feature = "serde")]
+struct MessageVisitor;
+
+#[cfg(feature = "serde")]
+impl<'de> Visitor<'de> for MessageVisitor {
+    type Value = Message;
+
+    fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+        formatter.write_str("an EIP-4361 formatted message")
+    }
+
+    fn visit_str<E>(self, value: &str) -> Result<Self::Value, E>
+    where
+        E: de::Error,
+    {
+        let message = match Message::from_str(value) {
+            Ok(message) => Ok(message),
+            Err(error) => Err(E::custom(format!("error parsing message: {error}"))),
```

Comment:
Sorry to be a pain, I didn't realise this is a recent language feature. Can we do something like this to avoid increasing the minimum required rust version?
```suggestion
            Err(error) => Err(E::custom(format!("error parsing message: {}", error))),
```